### PR TITLE
[RHOAIENG-4226] Add additional tracking for projects UI.

### PR DESCRIPTION
**Pull request by pilhuhn**
## Description

This adds additional Segment tracking. No visible UI changes are done.
This is recorded in https://issues.redhat.com/browse/RHOAIENG-4226

## How Has This Been Tested?

Click through projects UI, see events flow into Segment (or in dev-mode, watch browser console)

## Test Impact

UI behaviour is not changed.

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by andrewballantyne — APPROVED** [review]

**Comment by lucferbux** [comment]
Also, @pilhuhn is there any jira issue to link to the PR?

**Comment by pilhuhn** [comment]
> Also, @pilhuhn is there any jira issue to link to the PR?

See PR title :)  RHOAIENG-4226

File: frontend/src/utilities/segmentIOUtils.tsx [hunk 1]
```diff
@@ -1,6 +1,22 @@
 import { TrackingEventProperties } from '~/types';
 import { DEV_MODE } from './const';
 
+// The following is like the original method below, but allows for more 'free form' properties.
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const fireTrackingEventRaw = (eventType: string, properties?: any): void => {
```
[the remark this answers is not in the document]
Comment (pilhuhn) [reply frontend/src/utilities/segmentIOUtils.tsx:6]: Looks like the linter statement only caught half of it. Should be happy now

**Review by lucferbux — APPROVED** [review]
/lgtm

I've tested the events and they are working just fine, I'm gonna approve it, my only concern is wethere this raw events are well categorized in segment, I haven't tried that

**Pull request merged**